File: src/algorithms/are-occurrences-equal.js
/**
 * Given a string s, returns true if all characters that appear in s
 * have the same number of occurrences.
 * 
 * @param {string} s - A string of characters
 * @returns {boolean} - Returns true if all chars in s occur the same amount
 */
function areOccurrencesEqual (s) {
  const charCount = {};
  let comparisonNum = 0;
  for (let i = 0; i < s.length; i++) {
    charCount[s[i]] = (charCount[s[i]] || 0) + 1;
    if (i + 1 === s.length) {
      comparisonNum = charCount[s[i]];
    }
  }
  for (const [key, value] of Object.entries(charCount)) {
    if (value !== comparisonNum) {
      return false;
    }
  }
  return true;
}

File: src/algorithms/are-occurrences-equal.py
def areOccurrencesEqual(s):
    """Given a string s, return true if s is a good string, or false otherwise.
    A string s is good if all the characters that appear in s have the same 
    number of occurrences (i.e., the same frequency).

    Args:
        s (str): A string of lowercase English letters.

    Returns:
        Boolean: Returns True if all characters in s have the same frequency.
    """
    chars = {}
    for char in s:
        if char in chars:
            chars[char] += 1
        else:
            chars[char] = 1
    checkNumber = chars[s[0]]
    for val in chars.values():
        if val != checkNumber:
            return False
    return True
